test/e2e_tests/specs/Reply/reply.spec.ts

```diff
@@ -0,0 +1,353 @@
+import {ConfirmModal} from 'test/e2e_tests/pageManager/webapp/modals/confirm.modal';
+import {test as baseTest, expect} from 'test/e2e_tests/test.fixtures';
+import {getAudioFilePath, getTextFilePath, getVideoFilePath, shareAssetHelper} from 'test/e2e_tests/utils/asset.util';
+import {getImageFilePath} from 'test/e2e_tests/utils/sendImage.util';
+import {removeCreatedUser} from 'test/e2e_tests/utils/tearDown.util';
+import {createGroup, loginUser} from 'test/e2e_tests/utils/userActions';
+
+const test = baseTest.extend<{userA: User; userB: User}>({
+  userA: async ({api}, use) => {
+    const userA = getUser();
+    await api.createPersonalUser(userA);
+    await use(userA);
+    await removeCreatedUser(api, userA);
+  },
+  userB: async ({api}, use) => {
+    const userB = getUser();
+    await api.createPersonalUser(userB);
+    await use(userB);
+    await removeCreatedUser(api, userB);
+  },
+});
+
+const createPagesForUser = async (browser: Browser, user: User, options?: {openConversationWith?: User}) => {
+  const context = await browser.newContext();
+  const page = await context.newPage();
+  const pageManager = PageManager.from(page);
+
+  await pageManager.openMainPage();
+  await loginUser(user, pageManager);
+  const {pages, modals} = pageManager.webapp;
+
+  await modals.dataShareConsent().clickDecline();
+
+  if (options?.openConversationWith) {
+    await pages.conversationList().openConversation(options.openConversationWith.fullName);
+  }
+
+  return pages;
+};
+
+test.describe('Reply', () => {
+  test.beforeEach(async ({api, userA, userB}) => {
+    await api.connectUsers(userA, userB);
+  });
+
+  test(
+    'I should not be able to reply to a ping',
+    {tag: ['@TC-8038', '@regression']},
+    async ({browser, userA, userB}) => {
+      const pages = await createPagesForUser(browser, userA, {openConversationWith: userB});
+
+      await pages.conversation().sendPing();
+
+      const ping = pages.conversation().systemMessages.last();
+      await expect(pages.conversation().systemMessages.last()).toContainText('pinged');
+
+      await pages.conversation().systemMessages.last().hover();
+      await expect(ping.getByTestId('do-reply-message')).not.toBeAttached();
+    },
+  );
+
+  test(
+    'I should not be able to reply to timed messages',
+    {tag: ['@TC-8039', '@regression']},
+    async ({browser, userA, userB}) => {
+      const pages = await createPagesForUser(browser, userA, {openConversationWith: userB});
+
+      await pages.conversation().sendTimedMessage('Gone in 10s');
+      const message = pages.conversation().getMessage('Gone in 10s');
+      await message.hover();
+
+      await expect(message.getByTestId('do-reply-message')).not.toBeAttached();
+    },
+  );
+
+  test(
+    'I want to see a placeholder text as quote when original message is not available anymore',
+    {tag: ['@TC-2994', '@regression']},
+    async ({browser, userA, userB}) => {
+      const [userAPages, userBPages] = await Promise.all([
```

Comment:
This could also be abstracted away in a helper function, which takes an array of users, to be able to be easily shared with other functions.